master/internal/rm/agentrm/resources.go

```diff
@@ -59,8 +59,11 @@ func (c containerResources) Start(
 	spec.ExtraEnvVars[sproto.ResourcesTypeEnvVar] = string(sproto.ResourcesTypeDockerContainer)
 	spec.UseHostMode = rri.IsMultiAgent
 	spec.Devices = c.devices
-	// Write the real DET_UNIQUE_PORT_OFFSET value now that we know which devices to use.
-	spec.ExtraEnvVars["DET_UNIQUE_PORT_OFFSET"] = strconv.Itoa(tasks.UniquePortOffset(spec.Devices))
+
+	for portName, port := range rri.Ports {
+		spec.Environment.RawPorts[portName] = port
+		spec.ExtraEnvVars[portName] = strconv.Itoa(port)
+	}
```

Comment:
k8s and slurm need to replicate this logic in their corresponding resources `K8sPodResources` and `SlurmJobResources` or whatever that one is called

Reply:
or better yet.. just lift this up into the allocation actor

Reply:
could you also make a ticket for 'dedup resources start logic' this is getting out of hand

Reply:
yes will do.

Reply:
https://hpe-aiatscale.atlassian.net/browse/DET-9143